### ISSUE TITLE
Universal BlurHash widget

### ISSUE THREAD

**Issue by zs-dima** [fluttercommunity/flutter_blurhash#51]
Remote resource to download image is very specific way to use BlurHash widget.
Images could be loaded in many different ways using different protocols.
Could be nice to make BlurHash widget universal.

**Comment by dhikshith12** [comment]
Yes, to solve this i had build a package [blurhash_ffi](https://pub.dev/packages/blurhash_ffi) that takes in any ImageProvider subclass, which includes AssetImage, NetworkImage, MemoryImage, etc. it has both encoding and decoding which runs on a separate isolate with native C bindings found in original [blurha.sh](https://blurha.sh/) implementation.

It has Same API as flutter_blurhash and blurhash_dart, which is nice because you don't have to change much in your code.

Checkout if you are interested!
